presto-main/src/main/java/com/facebook/presto/sql/analyzer/PredicateStitcher.java

```diff
@@ -81,21 +82,24 @@ protected Node visitQuerySpecification(QuerySpecification node, Void context)
     }
 
     @Override
-    protected Node visitJoin(Join node, Void context)
+    protected Node visitJoin(Join node, PredicateStitcherContext context)
     {
         Relation rewrittenLeft = (Relation) process(node.getLeft(), context);
         Relation rewrittenRight = (Relation) process(node.getRight(), context);
         return new Join(node.getType(), rewrittenLeft, rewrittenRight, node.getCriteria());
     }
 
     @Override
-    protected Node visitAliasedRelation(AliasedRelation node, Void context)
+    protected Node visitAliasedRelation(AliasedRelation node, PredicateStitcherContext context)
     {
-        return new AliasedRelation((Relation) process(node.getRelation(), context), node.getAlias(), node.getColumnNames());
+        context.setCreateAlias(false);
```

Comment:
@jainxrohit , would you mind providing an example or more details why we need to set context before & after we visit a Relation child of AliasedRelation? If I remember correctly, aliased Table node is not AliasedRelation.

Sorry, my local env is kind of broken again and I'm fixing. So didn't get chance to try myself before sending out this review asap.

Reply:
I think the reason we need to set the context is that we only want to create the new alias node if it is not present. Please refer to the visitTable (where this context is used) and the new test added in this PR.

Reply:
If I understand right, if there's an alias node it will be visited by this method and we do need to reconstruct the node in that case.

I don't think the new test case added has an alias node, if I debug correctly. Can you confirm?

Reply:
Talked and debugged with Ge offline, changes are working as expected.